src/errors/InvalidIdError.ts

```diff
@@ -0,0 +1,8 @@
+import { Annotated } from './Annotated';
+
+export class InvalidIdError extends Error implements Annotated {
+  specReferences = ['http://hl7.org/fhir/R4/datatypes.html#id'];
+  constructor(public badId: string) {
+    super(`The string "${badId}" does not represent a valid FHIR id.`);
```

Comment:
Can we provide some more guidance here?  Perhaps repeating what the FHIR doc says.  So maybe something like:

> The string "${badId}" does not represent a valid FHIR id.  FHIR ids may contain any combination of upper- or lower-case ASCII letters ('A'..'Z', and 'a'..'z'), numerals ('0'..'9'), '-' and '.', with a length limit of 64 characters.